### ISSUE TITLE
Set-CsOnlinePstnUsage & New-CsOnlineVoiceRoute failing with MicrosoftTeams v3.1.0

### ISSUE THREAD

**Issue by M0rfic** [MicrosoftDocs/OfficeDocs-SkypeForBusiness#8431]
Hi,

Just removed 3.0.0 and installed v3.1.0 and some cmdlets that were previously running fine now fail. It looks like a bug, but I'd like to make sure I'm not missing something as the docs don't seem to have changed. Googling around I found [someone else](https://matrixpost.net/teams-powershell-several-cmdlet-doesnt-work-errounable-to-cast-object-of-type-system-management-automation-pslistmodifier-to-type-system-string/) having a similar problem, but nothing else. So far the same thing happens for (please note that I edited out the sensitive information):

- [`Set-CsOnlinePstnUsage`](https://docs.microsoft.com/en-us/powershell/module/skype/set-csonlinepstnusage?view=skype-ps)

```
PS D:\> Set-CsOnlinePstnUsage -Identity Global -Usage @{ add = "XXX" } -ErrorAction Stop
Set-CsOnlinePstnUsage : Unable to cast object of type 'System.Management.Automation.PSListModifier' to type 'System.String'.
At line:1 char:1
+ Set-CsOnlinePstnUsage -Identity Global -Usage @{ add = "XXX" ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Set-CsOnlinePstnUsage], InvalidCastException
    + FullyQualifiedErrorId : System.InvalidCastException,Microsoft.Rtc.Management.Internal.SetOnlinePstnUsageCmdlet
```

- [`New-CsOnlineVoiceRoute`](https://docs.microsoft.com/en-us/powershell/module/skype/new-csonlinevoiceroute?view=skype-ps)

```
PS D:\> New-CsOnlineVoiceRoute -Identity "YYY" -Priority 1 -NumberPattern ".*" -OnlinePstnGatewayList @{ add = "some domains" }  -OnlinePstnUsages "XXX" -ErrorAction Stop
New-CsOnlineVoiceRoute : Unable to cast object of type 'System.Management.Automation.PSListModifier' to type 'System.String'.
At line:1 char:1
+ New-CsOnlineVoiceRoute -Identity "YYY" -Priority 1 -NumberP ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [New-CsOnlineVoiceRoute], InvalidCastException
    + FullyQualifiedErrorId : System.InvalidCastException,Microsoft.Rtc.Management.Internal.NewOnlineVoiceRouteCmdlet
```

**Comment by scanum** [comment]
Thank you for submitting feedback.
I think the best way forward is if you open a service ticket in your tenant so this can get resolved ASAP. Based on the outcome let me know if it is something that can be called out in the docs.
Please follow this link to contact support for business products:https://docs.microsoft.com/office365/admin/contact-support-for-business-products
Please keep us posted here on the resolution so we can feed whatever information you discover into the content.

**Comment by JensTrier** [comment]
@M0rfic not anything you do wrong, it also reproes for me :-) 

In general I agree on the support route, since support typically knows the bigger picture and can escalate as needed. For this one I can say we are investigating the issue and no need to do anything more on your part. Using this route is more geared to feedback on the documentation itself.

I have a vested interest in some of the PS cmdlets and that's why I'm replying there once in a while :-)

**Comment by M0rfic** [comment]
@scanum  & @JensTrier 

Regarding the issue itself, looks like the bug has been resolved in the newly released 3.1.1 version. Below you find the release notes from the [PowerShell Gallery page](https://www.powershellgallery.com/packages/MicrosoftTeams/3.1.1), at the end of the Package Details section:

> Release Notes
>**3.1.1-GA** (The project - MicrosoftTeams contains these changes so far for this release)
>
>- Performance improvements for Import-Module.
>- PSListModifier deserialization issue has been fixed.

Just tested this locally and everything is back to normal. I'm not sure if you guys need to update anything in the docs, except maybe state that the cmdlets are broken in 3.1.0. Anyway don't want to overstep in your garden, so you do you 😄 

----

Regarding the route follow for such issues, I understand this repo is only for docs related issues and based on your comment I take it there is no dedicated issue tracker where we can reach directly someone from the tech team, and we should always go through tenant support. Doesn't seem ideal, but it is what it is, noted, thanks.

----

Nonetheless, thanks for your time and support.

Kind regards

**Comment by JensTrier** [comment]
Yeah, was waiting for the release notes to be published to let you know that we fixed it :-)
